ComplementaryScripts/modelCuration/mapKEGGID.m

```diff
@@ -0,0 +1,275 @@
+empties = find(cellfun('isempty',xref_rxnKEGG(:,1)));
+xref_rxnKEGG(empties,:) = [];
+empties = find(~cellfun('isempty',xref_rxnKEGG(:,4)));
+xref_rxnKEGG(empties,:) = [];
+
+%add rxnKEGGID into model
+[~,idx] = ismember(xref_rxnKEGG(:,2),model.rxns);
+idx = idx(idx~=0);
+
+for i = 1:length(idx)
+    model.rxnKEGGID(idx(i)) = xref_rxnKEGG(i,1);
+end
+
+%correction for MNXref error in MNXR96123/R00124#2
+idx = find(ismember(model.rxnMetaNetXID,'MNXR96123'));
+model.rxnKEGGID(idx) = {'R00124'};
+
+%map rxnMetaNetXID to rxnMetaCycID via function mapIDsViaMNXref, then map rxnMetaCycID to rxnKEGGID
+xref_rxnMetaCyc = mapIDsViaMNXref('rxns',model.rxnMetaNetXID,'MetaNetX','MetaCyc');
+xref_rxnMetaCyc(:,2) = model.rxns; %match rxnMetaCycID with model.rxns
+xref_rxnMetaCyc(:,3) = model.rxnMetaNetXID; %match rxnMetaCycID with model.rxnMetaNetXID
+xref_rxnMetaCyc(:,4) = model.rxnKEGGID; %match rxnMetaCycID with model.rxnKEGGID
+empties = find(cellfun('isempty',xref_rxnMetaCyc(:,1)));
+xref_rxnMetaCyc(empties,:) = [];
+
+%load metaCycRxns.mat via getRxnsFromMetaCyc from RAVEN directory
+MetaCyc_rxnInfo = getRxnsFromMetaCyc('..RAVEN/external/metacyc');
+
+%Input rxnKEGGID into xref array
+for i = 1:length(xref_rxnMetaCyc)
+    if isempty(xref_rxnMetaCyc{i,4})
+        idx = find(ismember(MetaCyc_rxnInfo.rxns,xref_rxnMetaCyc(i,1)));
+        if idx~=0
+            if ~isempty(MetaCyc_rxnInfo.rxnMiriams{idx})
+                temp = cell2mat(MetaCyc_rxnInfo.rxnMiriams(idx));
+                idx2 = find(ismember(temp.name,'kegg.reaction'));
+                if ~isempty(idx2)
+                    xref_rxnMetaCyc(i,5) = temp.value(idx2);
+                end
+            end
+        end
+    end
+end
+
+if size(xref_rxnMetaCyc,2) == 5
+    %only retain mapped rxnKEGGID if model.rxnKEGGID is empty
+    empties = find(cellfun('isempty',xref_rxnMetaCyc(:,5)));
+    xref_rxnMetaCyc(empties,:) = [];
+    empties = find(~cellfun('isempty',xref_rxnMetaCyc(:,4)));
+    xref_rxnMetaCyc(empties,:) = [];
+    
+    %add rxnKEGGID into model
+    [~,idx] = ismember(xref_rxnMetaCyc(:,2),model.rxns);
+    idx = idx(idx~=0);
+    
+    for i = 1:length(idx)
+        model.rxnKEGGID(idx(i)) = xref_rxnMetaCyc(i,5);
+    end
+else
+    xref_rxnMetaCyc(:) = []; %erase all cell array, since no KEGGID matched and added
+end
+
+%map rxnMetaNetXID to rxnSEEDID via function mapIDsViaMNXref, then map rxnSEEDID to rxnKEGGID
+xref_rxnSEED = mapIDsViaMNXref('rxns',model.rxnMetaNetXID,'MetaNetX','SEED');
+xref_rxnSEED(:,2) = model.rxns; %match rxnSEEDID with model.rxns
+xref_rxnSEED(:,3) = model.rxnMetaNetXID; %match rxnSEEDID with model.rxnMetaNetXID
+xref_rxnSEED(:,4) = model.rxnKEGGID; %match rxnSEEDID with model.rxnKEGGID
+empties = find(cellfun('isempty',xref_rxnSEED(:,1)));
+xref_rxnSEED(empties,:) = [];
+
+%load reactions.tsv from modelSEEDDatabase
+temp = what('ModelSEEDDatabase\Biochemistry'); %require cloned/downloaded GitHub repo ModelSEEDDatabase to be added to MATLAB path
```

Comment:
Consider firstly check if the file exist, if not then download directly using `git('clonehttps://github.com/ModelSEED/ModelSEEDDatabase')`  here. If it is downloaded here, remove the file at the end. Otherwise, this requirement should be included in the beginning of this function.

Reply:
Solved in b8a5d3283c45e15a654eb5e89e91570a7406ff61